docs/user/max_resource_usage.md

```diff
@@ -0,0 +1,141 @@
+# Maximize resource usage
+
+To accelerate running an experiment, we want to maximize the usage of resources
+(CPUs or GPUs).
+
+## Maximize CPU utilization
+
+The most compute-intensive part in training a policy is sampling. Garage uses
+[`Worker`](sampling.html#worker) to abstract a CPU to perform a rollout and uses
+[`Sampler`](sampling.html#sampler) to manage them. To maximize CPU utilization,
+we only need to choose proper `Sampler` and `Worker` when setting up
+`LocalRunner`.
+
+In the following tests, we will use [`examples/torch/trpo_pendulum.py`](https://github.com/rlworkgroup/garage/blob/master/examples/torch/trpo_pendulum.py)
+as example experiment file.
+
+### `LocalSampler` and `DefaultWorker`
+
+First, we use the most basic config, `LocalSampler` and `DefaultWorker`. The
+sampler will run workers in the same process.
+
+```py
+from garage.sampler import LocalSampler, DefaultWorker
+
+    ...
+    runner.setup(algo, env, sampler_cls=LocalSampler, worker_class=DefaultWorker)
+    ...
+```
+
+With `top` command, in my environment (4 cores), we can see the CPU usage is
+about 300%.
+
+```sh
+...
+  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND
+14610 ruofu     20   0 4555092 558680 268692 R 301.0  6.9   6:06.37 python examples
+...
+```
+
+### Use `VecWorker`
+
+To reduce the overhead in sampling, we use `VecWorker` to run multiple
+environments in one step.
+
+```py
+from garage.sampler import LocalSampler, VecWorker
+
+    ...
+    runner.setup(algo, env, sampler_cls=LocalSampler, worker_class=VecWorker)
+    ...
+```
+
+And the CPU usage is about 350%.
+
+```sh
+...
+  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND
+11902 ruofu     20   0 4696076 709944 268424 R 351.2  8.7   2:56.95 python examples
+...
+```
+
+### Use `RaySampler`
+
+Though the CPU usage is increased, it still run in one process. We can
+parallelize sampling across CPUs using `RaySampler` to maximize the CPU usage:
```

Comment:
Please detail that you can use `RaySampler` to run sampling on any environment ray supports (e.g. clusters, k8s, etc.), not just the local machine.